### DISCUSSION TITLE
Detox Suppress Firebase In-App Messaging React Native

### DISCUSSION THREAD

**Discussion by vtnx8585** [wix/Detox#3403] [Q&A]
Problem:

I'm working on a React Native app in which we have Detox testing that recently was working correctly. Recently we added Firebase In-App Messaging and it started breaking the tests because the firebase messages visually blocks the app screens making Detox testing not able to tap or verify any component. We have already tried using `await inAppMessaging().setMessagesDisplaySuppressed(true);` but doesn't work at all to suppress this messages as per documentation [https://rnfirebase.io/in-app-messaging/usage]. Using the workaround of just making Detox tap 'ok' in firebase message doesn't solve the issues because you can't know when exactly the message will show up, plus is not a solid solution if the main purpose of our testing is not to check Firebase In-App messages, we should just suppress it.

Questions:

1) Has anyone ran Detox successfully with Firebase In-App Messages and if so how did you made it work?
2) Could the `inAppMessaging().setMessagesDisplaySuppressed(true)` not working because of the many reloads Detox before starting each suite test?

**Comment by portyu9** [comment]
Hi @vtnx8585 

I would disable/suppress Firebase In-App Messaging as part of the application's E2E startup rather than teaching Detox to dismiss Firebase's UI.

A clean approach is to launch with an E2E flag:

```js
await device.launchApp({
  newInstance: true,
  launchArgs: {
    e2e: 1,
  },
})
```

Then, during application startup, detect that E2E flag and disable/suppress Firebase In-App Messaging before messages can appear.

The timing is important.

If suppression only happens once outside the application's normal bootstrap, restarting/reloading the app can initialize Firebase again and make the behavior look intermittent.

Prefer:

```text
E2E launch
 -> detect E2E flag
 -> initialize Firebase
 -> suppress FIAM
 -> render application
```

on every relevant application startup.

Even better, if Firebase In-App Messaging itself is not what the test is supposed to cover, disable it entirely in your dedicated E2E build/configuration.

That is much more deterministic than having every Detox test dismiss a Firebase popup.

Please mark this answer as accepted if it helped, thank you.
